File: GTP2/ej5.cpp
#include <cstring>
#include <iostream>
#include "ej5_lib.h"

using namespace std;

int main(int argc, char *argv[]) {
    if (argc != 2 or strlen(argv[1]) > 10) {
        cout << "Uso: " << argv[0] << " STRING" << endl;
        cout << "STRING de largo 10" << endl;
        exit(1);
    }
    
    char *s = new char[strlen(argv[1])];
    s = argv[1];

    mayusculizar(s);

    cout << s << endl;

    return 0;
}

File: GTP2/ej5.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	if len(os.Args) != 2 || len(os.Args[1]) > 10 {
		fmt.Printf("Uso: %s CADENA\nCADENA de largo <= 10\n", os.Args[1])
		os.Exit(1)
	}

	fmt.Println(mayusculizar(&os.Args[1]))
}

func mayusculizar(s *string) string {
	return strings.ToUpper(*s)
}
